### ISSUE TITLE
Feature request: search gist content

### ISSUE THREAD

**Issue by kergoth** [hackjutsu/Lepton#82]
It'd be nice if the search covered more than just description/filename, as quite often what I'm looking for is a command inside a shell script or something :)

**Comment by hackjutsu** [comment]
@kergoth 

Thanks for feedback. Lepton uses **"lazy loading"** for gists, which means the gist content is not fetched until users click on it. This mechanism limits the local search's capability, since it can only search for what has been downloaded. 

The next release(the coming weekend) will have a big improvement on the search quality, like fuzzy search and unicode support. But as I just mentioned, I haven't come up with a good solution for content search yet.

Any suggestions or PR will be welcome:)

**Comment by kergoth** [comment]
Fair enough, I figured the fetching was the issue, as I'm guessing the gist api doesn't provide content search. Thanks for the quick reply, I look forward to playing with Lepton more, you've done nice work so far. It's a shame github's built in gist UI is so limited, particularly for those of us with a *lot* of gists :)

**Comment by AStupidBear** [comment]
+1

**Comment by DestructiveInfluence** [comment]
Still a highly desired feature.

**Comment by supremeo** [comment]
+1 this drastically limits the functionality of this otherwise super cool app.

**Comment by dmwyatt** [comment]
I'd prefer a toggle or button that went ahead and forced a download of all gists so that we could do content search.

In other words, search continued to work like it does currently, but we could force it to go ahead and download everything so it could be searched.

**Comment by zhezh** [comment]
Since gists are mostly text, it will not take much storage space to download them all. I guess full text search on the cached gists would be a good choice.

**Comment by palexof** [comment]
I just cannot beleive there is not a single tool which supports searching through the contents of your gists. How hard should it be to download all gists throught the GutHub API, put them in an sqllite file and do some good old information retreival?!?

**Comment by ikaygorodov** [comment]
+1 This feauter is really required. I would also agree that text files should not take to much space.

**Comment by hackjutsu** [comment]
Implemented in the current v2 prerelease line. Lepton now indexes snippet metadata plus downloaded file content; complete global content search is available when snippet.downloadAll is enabled before sync. See the Search FAQ for the current behavior. Closing this as resolved in v2.0.0-beta.7.
